app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    for person in people:
+        name, age = person["name"], person["age"]
+        Person(name, age)
+
+    for person in people:
+        name = person["name"]
+        spouse = person.get("wife") or person.get("husband")
+        if spouse:
+            setattr(Person.people[name],
+                    "wife" if "wife" in person else "husband",
+                    Person.people[spouse])
+
+    return list(Person.people.values())
```

Comment:
The function `create_person_list` modifies the class-level dictionary `Person.people`. If this function is called multiple times, it will accumulate people from all calls, which might not be the intended behavior. Consider resetting `Person.people` at the start of the function or using an instance-level dictionary.